RELEASING.md

```diff
@@ -0,0 +1,69 @@
+# Release Checklist
+
+These steps assume that you've checked out the `scale-info` repository and are in the root directory of it.
+
+We also assume that ongoing work done is being merged directly to the `master` branch.
+
+1.  Ensure that everything you'd like to see released is on the `master` branch.
+
+2.  Create a release branch off `master`, for example `release-v0.17.0`. Decide how far the version needs to be bumped based
+    on the changes to date.
+
+3.  Check that you're happy with the current documentation.
+
+    ```
+    cargo doc --open
+    ```
+
+    If there are minor issues with the documentation, they can be fixed in the release branch.
+
+4.  Bump the crate versions in `./Cargo.toml` and `./derive/Cargo.toml` to whatever was decided in step 2 (basically a find and
+    replace from old version to new version in this file should do the trick).
+
+5.  Ensure the `Cargo.lock` file is up to date.
+
+    ```
+    cargo generate-lockfile
```

Comment:
why? this is just library so `Cargo.lock` shouldn't matter?

Reply:
Ohyeah whoops; carried over from our Subxt one!